packages/components/src/Chip/Chip.css

```diff
@@ -111,3 +112,76 @@
 .chip.disabled .suffix {
   background-color: var(--color-disabled);
 }
+
+.chipHeading {
+  flex-shrink: 0;
+}
+
+.chipHeading,
+.chipLabel {
+  position: relative;
+  overflow: hidden;
+}
+
+.chipHeading > p,
+.chipLabel > p {
+  overflow: hidden;
+  white-space: nowrap;
+}
+
+.truncateGradient {
+  position: absolute;
+  top: 0;
+  right: 0;
+  width: var(--space-base);
+  height: 100%;
+  background: linear-gradient(
+    to left,
+    var(--chip-base-bg) 25%,
```

Comment:
Seeing all these repeated declarations makes me wonder if we can clean up our component-level "tokens"?

If you look at [Button](https://github.com/GetJobber/atlantis/blob/817bc0911d3f0f806a035559d6ab34954f4b5a31/packages/components/src/Button/Button.css#L59-L81) we avoid a lot of repetitive declarations by updating the variables on a state level.

Reply:
oh definitely

this is just me trying to get this working at all before optimizing it. there's some weird stuff happening that I need to figure out if we can fix or not with this approach

Reply:
k got this working and followed up with a similar approach with a single gradient definition that looks at a variable that we swap around depending on the variation/state